docs/source/security/authentication.mdx

```diff
@@ -226,20 +257,16 @@ const typeDefs = `
     banned: Boolean @auth(requires: ADMIN)
     canPost: Boolean @auth(requires: REVIEWER)
   }
-`
+`;
 ```
 
 The `@auth` directive can be called directly on the type, or on the fields if you want to limit access to specific fields as shown in the example above. The logic behind authorization is hidden away in the directive implementation.
 
-One way of implementing the `@auth` directive is by constructing your schema using the [`schemaTransforms` option of the `makeExecutableSchema`](https://www.graphql-tools.com/docs/schema-directives/) function from the [graphql-tools](https://www.graphql-tools.com/) project. You can read the `graphql-tools` docs for details on how to write `schemaTransforms`; then just create your server with
-```js
+One way of implementing the `@auth` directive is by constructing your schema using the [`schemaTransforms` option of the `makeExecutableSchema`](https://www.graphql-tools.com/docs/schema-directives/) function from [graphql-tools](https://www.graphql-tools.com/), like so:
```

Comment:
This option was removed from `makeExecutableSchema` a year ago.

Maybe replace this paragraph with something like:

Now that you've added directives to your schema, you need to make all of your resolvers check the role on the context against the role specified by the directive. One way to do that is to use the `mapSchema` function from the third-party `@graphql-tools/utils` to transform every resolver in your schema. Check out [their example of adding directive-based permission checks](https://www.graphql-tools.com/docs/schema-directives#enforcing-access-permissions).

(I don't think we should feel the need to actually write the code that uses their API in our docs that we then need to keep up to date.)

Reply:
Oh, good to know! I tweaked your suggested paragraph above slightly; let me know if this sounds good to you:

> The `@auth` directive can be called directly on specific fields or on a type. This nicely hides the authorization logic behind the directive implementation.
>
>With your `@auth` directive in place, your resolvers now need to check the user's role against the role specified by the directive. One way to do this is using the `mapSchema` function from the [`@graphql-tools/utils`](https://www.npmjs.com/package/@graphql-tools/utils) package to transform every resolver in your schema. For an example of setting up directive-based permission checks, see the [`mapSchema` docs](https://www.graphql-tools.com/docs/schema-directives#enforcing-access-permissions).

(I also removed the code snippet)

Reply:
Sounds good